lib/array.rb

```diff
@@ -0,0 +1,26 @@
+class Array
+  def custom_inject(*args, &block)
+    operation, injector = retrieve_args(*args)
+    injector ||= shift # i.e: self.shift (RuboCop is not being friendly!)
```

Comment:
Very true, but it seemed weird to me to see ‘shift’ without something in front of it (bad Java perspective, perhaps?).